packages/flutter_tools/test/general.shard/windows/visual_studio_test.dart

```diff
@@ -146,73 +136,80 @@ void main() {
     VisualStudio visualStudio;
```

Comment:
Oops. I'm not sure why I was declaring this here.

I would recommend going the other way from what this patch is doing (removing almost all of the test-function-local creations) and instead remove this variable and the setup creating it; if you want to simplify the local sites, you could have an explicit helper function that returns the simple version you're currently creating. In general, I think it's a confusing practice in tests to have the primary object under test have a scope outside the test function, because it becomes hard to reason about what its state is. Does some helper being called at any point in the tests affect it? No way of knowing without reading all of them. And then it's easy for someone to add later. https://abseil.io/tips/122 has a good discussion of this.

Reply:
I've updated this test to no longer use any unnecessary top levels, by exposing as `visibleForTesting` certain members of VisualStudio. I also hoisted the helpers out of main to follow better Dart testing practices.

In general, I'm not sure how to square the very good advice you've given me with the current testing practices we're using. Maybe we just need to keep working on making them better?

Reply:
> In general, I'm not sure how to square the very good advice you've given me with the current testing practices we're using. Maybe we just need to keep working on making them better?

I think so, yes. `globals` could be viewed as the complex test fixture antipattern on steroids, since it's doing basically the same thing but beyond the scope of tests, so moving away from that is probably the biggest thing that can currently be done to move toward the advice linked above. Longer term moving away from lots of environment in test fixtures might be nice, but that's a smaller problem. The one thing I think is worth strongly avoiding while making the incremental steps is having the primary test object be in the fixture, which is why I flagged that in particular.